File: iosApp/src/iosAppMain/apple/features/characters/ui/CharacterViewData.swift
import Foundation
import shared

public struct CharacterViewData: Identifiable {
    public let id: Int32
    let name: String
    let status: String
    let imageUrl: URL?
    
    public init(character: Character) {
        id = character.id
        name = character.name
        status = character.status
        imageUrl = URL(string: character.imageUrl)
    }
}

File: androidApp/src/main/java/za/co/izakvdhoven/kmmplayground/android/features/characters/ui/CharacterViewData.kt
package za.co.izakvdhoven.kmmplayground.android.features.characters.ui

import android.net.Uri
import androidx.core.net.toUri
import za.co.izakvdhoven.kmmplayground.features.characters.domain.models.Character

data class CharacterViewData(
    val name: String,
    val status: String,
    val imageUrl: Uri
) {
    constructor(character: Character) : this(
        name = character.name,
        status = character.status,
        imageUrl = character.imageUrl.toUri()
    )
}